### PR TITLE
Upgrade internal bitcoind version

### PR DESCRIPTION
**Pull request by ben-kaufman**
Update the packaged `bitcoind` version to `0.21.1` and add an upgrade mechanism for this and future updates

### CODE REVIEW

File: pyinstaller/specterd.spec [hunk 1]
```diff
@@ -33,16 +33,16 @@ elif platform.system() == 'Linux':
                 break
     packaged_software_datas = [
         ('./torbrowser/tor-browser-linux64-10.0.15_en-US.tar.xz', 'torbrowser'),
-        ('./bitcoind/bitcoin-0.21.0-x86_64-linux-gnu.tar.gz', 'bitcoind'),
-        ('./bitcoind/bitcoin-0.21.0-arm-linux-gnueabihf.tar.gz', 'bitcoind'),
+        ('./bitcoind/bitcoin-0.21.1-x86_64-linux-gnu.tar.gz', 'bitcoind'),
+        ('./bitcoind/bitcoin-0.21.1-arm-linux-gnueabihf.tar.gz', 'bitcoind'),
```
Comment (stepansnigirev) [inline pyinstaller/specterd.spec:37]: I think we could package only one version of bitcoin core here depending on `platform.processor()`

Comment (ben-kaufman) [reply pyinstaller/specterd.spec:37]: But then we'll have to make 2 different Linux builds on 2 different Linux machines.

Comment (stepansnigirev) [reply pyinstaller/specterd.spec:37]: But we need to do it anyway. Currently we release only for x86_64, for arm we need another build machine

Comment (ben-kaufman) [reply pyinstaller/specterd.spec:37]: Right, my bad. I'll update it

Comment (ben-kaufman) [reply pyinstaller/specterd.spec]: Updated

File: src/cryptoadvance/specter/specter.py [hunk 2]
```diff
@@ -685,7 +685,7 @@ def migrate_old_node_format(self):
                 self,
                 self.bitcoind_path,
                 "mainnet",
-                self.internal_bitcoind_version,
+                "0.20.1",
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/specter.py:688]: Should be 21.1, no?
```suggestion
                "0.21.1",
```

**Pull request merged**